Add PATs documentation (#108)
trento-project/docs: trento/adoc/trento-personal-access-tokens.adoc

include::product-attributes.adoc[]

[[sec-trento-personal-access-tokens]]

== Personal Access Tokens
:revdate: 2025-11-05

{trento} allows users to create Personal Access Tokens (PATs) for authentication and authorization purposes in third party integrations with the APIs exposed by its components:

 * link:https://www.trento-project.io/web/swaggerui/[Web]
 * link:https://www.trento-project.io/wanda/swaggerui/[Wanda]

=== About Personal Access Tokens

Personal Access Tokens are an alternative to using username/password pairs for authentication to {trento} APIs and are intended to access resources on behalf of the token owner.

A Personal Access Token has the same capabilities to access resources and perform actions on those resources that the owner of the token has.

[NOTE]
====

> If user permissions are changed or the user is deleted or locked, the capabilities of the Personal Access Tokens owned by that user will be affected accordingly.

Reviewer: nit:
maybe @dmpop will highlight this, but usually, we don't use future form in docs (as far as I remember).
So in this case, we would have `by that user are affected accordingly`.
There are some few more future case forms during the text

====

=== Creating a Personal Access Token

After logging in to {trento} Web UI, navigate to your profile by clicking on your username in the top right corner and selecting *Profile* from the dropdown menu.

> 

Reviewer 2: ```suggestion
Log in to {trento} Web UI, click on your username in the top-right corner, and select *Profile* from the drop-down menu.

```

+
.Generate personal access token modal
image::trento-web-generate-pat.png[trento-web-generate-pat,scaledwidth=80.0%]
. Copy the generated token starting with `trento_pat_`. This is the only time the token will be displayed in full. If lost, a new key must be generated.
+
.Generated personal access token
image::trento-web-generated-pat.png[trento-web-generated-pat,scaledwidth=80.0%]

=== Using a Personal Access Token

Once a Personal Access Token has been created, it can be used to authenticate API requests by including it in the `Authorization` header as a Bearer token.

[source,bash]
----
$ curl -X GET "..." -H "Authorization: Bearer trento_pat_<rest_of_token>"
----

=== Deleting a Personal Access Token

A Personal Access Token can be deleted by clicking its contextual *Delete* button in the *Personal Access Tokens* section of the profile.

.Personal access tokens section
image::trento-web-pats-list.png[trento-web-pats-list,scaledwidth=80.0%]

Confirm the deletion

.Delete personal access token modal
image::trento-web-delete-pat-confirmation.png[trento-web-delete-pat-confirmation,scaledwidth=80.0%]

=== Notes for user admins

> User admins can delete tokens for other users but cannot create new ones on their behalf.

Reviewer 2: ```suggestion
User admins can delete tokens for other users. However, admins cannot create new tokens for other users.
```
